### PR TITLE
Handle branch protection on private repositories.

### LINKED ISSUE #37: Handle BP on free-tier private repos
Branch Protection is not enabled on free-tier private repos: https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches
GitHub responds with:
`GET https://api.github.com/repos/org/repo/branches/master/protection: 403 Upgrade to GitHub Pro or make this repository public to enable this feature. []`
Currently we error out of the whole installation/org.

Change to fail BP policy, notify text should include the message from GitHub "Upgrade to GitHub Pro or make this repository public to enable this feature."

### PR DESCRIPTION
**Pull request by jeffmendoza**
Fixes #37

### CODE REVIEW

File: pkg/policies/branch/branch.go [hunk 1]
```diff
@@ -212,6 +212,15 @@ func check(ctx context.Context, rep repositories, c *github.Client, owner,
 				ds[b] = details{}
 				continue
 			}
+			if rsp != nil && rsp.StatusCode == http.StatusForbidden {
+				// Protection not available
+				pass = false
+				text = text + "Branch Protection enforcement is configured in Allstar, however Branch Protection is not available on this repository. Upgrade to GitHub Pro or make this repository public to enable this feature.\n" +
+					"See: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches for more information.\n" +
+					"Alternatively, disable Branch Protection policy enforcement for this repository in Allstar configuration."
```
Comment (inferno-chromium) [inline pkg/policies/branch/branch.go]: Can remove this alternatively line, it is bad recommendation?

**Review by inferno-chromium — DISMISSED** [review]

File: pkg/policies/branch/branch.go (hunk 1 above)
Comment (jeffmendoza) [reply pkg/policies/branch/branch.go]: If folks don't want to pay for GitHub pro, they can't setup branch protection on private repos. So they probably don't want Allstar configured to bug them to turn it on.

Comment (inferno-chromium) [reply pkg/policies/branch/branch.go]: ok, then can frame like
"If this is not feasible, then disable ..."

Alternatively makes it think it is ok to do that. so slight language change, then can keep line. thanks for explaining

Comment (jeffmendoza) [reply pkg/policies/branch/branch.go]: Yes, sounds great. Applied. Please re-approve.

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**